lib/riemannx/connections/batch_queue.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Riemannx.Connections.BatchQueue do
```

Comment:
Not sure this belongs in the connection namespace. You could possibly turn it into a generic behaviour and make it a configurable option to set another module in its place. This allows people to control the queueing mechanism if they want to.

Opening it up for the namespace `Riemannx.Queues.Qex`

Reply:
The problem is that the API of this queue is very specific for the `Batch` module. It could even be a submodule inside `Riemannx.Connections.Batch` if you prefer, as it will only work for that use case.

Reply:
You're right, that sounds good.

Reply:
Thanks for the call!

Just for documentation purposes, due to the tightly coupled nature of this implementation to the `Batch` connection we agreed to nest the module inside `Riemannx.Connections.Batch` which will give it the namespace `Riemannx.Connections.Batch.Queue`

Reply:
Done